src/app/modules/startup/module.nim

```diff
@@ -374,26 +375,10 @@ method onFetchingFromWakuMessageReceived*[T](self: Module[T], backedUpMsgClock:
   if self.view.fetchingDataModel().allMessagesLoaded():
     self.view.setCurrentStartupState(newProfileFetchingSuccessState(currStateObj.flowType(), nil))
 
-proc prepareAndInitFetchingData[T](self: Module[T]) =
+method prepareAndInitFetchingData*[T](self: Module[T]) =
   # fetching data from waku starts when messenger starts
   self.view.createAndInitFetchingDataModel(listOfEntitiesWeExpectToBeSynced)
 
-proc delayStartingApp[T](self: Module[T]) =
```

Comment:
@igor-sirotin why did you remove this and other delay-related functions?

Reply:
Maybe I didn't get something, but it wasn't making sense to me.

Current code flow (in case of importing seed phrase) is this:
1. After `BiometricsState` we automatically switch to `ProfileFetchingState`
2. In this state we wait for profile to be fetched and allow the user to proceed to app with `Continue` button. Otherwise there's a 2min timeout in UI.

Reply:
> Current code flow (in case of importing seed phrase) is this....

Yes, that's true, and that's also how it was, but there was one part that is missing now. That's 30 seconds delay (displaying app loading screen) before a user gets to the `ProfileFetchingState`. 

That 30-second delay was on purpose there and you didn't face any UI issues cause most likely either your Status account is new and has only a single set of backed-up data or the most recent set of backed-up data was received first. In that case all is good, cause the logic just rejects older data that comes later. 

But imagine now scenario where you have many sets of backed-up data (we're backing up every 8 hours for a user who keeps the app running all the time and keeps those messages for 30 days) and now (let's keep just to contacts related data, contacts progress bar) you have the following backed-up contacts data on the network, considering that time T1 < T2 < T3:
- BU-T1-C5 -> backup done at time T1, 5 contacts were backed-up
- BU-T2-C10 -> backup is done at time T2, 10 contacts were backed-up (you've added 5 more contacts in the meantime)
- BU-T3-C3 -> backup is done at time T3, 3 contacts were backed-up (you've removed 7 contacts in the meantime)

Without that 30-second delay the only valid UI data will be if the backed-up message done at the moment T3 comes first, otherwise user may see the. following scenario:
- fetching contacts 0/10... then
- fetching contacts 3/10... then
- fetching contacts 4/5...
- fetching contacts 5/5... and it may look like all is fetched, but then the latest set of messages arrives and then
- fetching contacts 1/3...
- fetching contacts 3/3 - which is the latest and correct number of contacts

So because of that, cause we're not sure about the order we're receiving backed-up messages in, we have that 30-second delay, cause if the data are fetched at all that's usually done in the first 10 seconds then we're waiting some time more, and when we switch to `ProfileFetchingState` we're pretty sure that we've already received the latest backed-up messages (ofc we cannot be sure ever about it, but in very very high percentage of cases that should be good).

Potential scenario:

![4](https://github.com/status-im/status-desktop/assets/86303051/4973a753-951d-48ab-a129-69bd3f0c27bc)
![5](https://github.com/status-im/status-desktop/assets/86303051/91a5f614-a864-465e-9760-14ebcf634e76)

Reply:
Thank you @saledjenic for a detailed answer.

I understand now the intention. But I think this solution only masks the actual problem and it makes  the `ProfileFetchingState` almost useless. Because of the 30 seconds delay we never see the progress bars growing and don't understand what is actually happening during these 30 seconds.

I think we should fix this on status-go level.

-----

> ... cause we're not sure about the order we're receiving backed-up messages 

This is actually the root cause of the problem. 
Here's how we currently fetch the backup from store nodes:

https://github.com/status-im/status-go/blob/3f19972c8e69720d474da48b53afd65cd71ab0fd/protocol/messenger_mailserver.go#L258-L274

```go
func (m *Messenger) syncBackup() error {

	filter := m.transport.PersonalTopicFilter()
	if filter == nil {
		return errors.New("personal topic filter not loaded")
	}

	from, to := m.calculateMailserverTimeBounds(oneMonthDuration)

	batch := MailserverBatch{From: from, To: to, Topics: []types.TopicType{filter.ContentTopic}}
	ms := m.getActiveMailserver(filter.ChatID)
	err := m.processMailserverBatch(*ms, batch)
	if err != nil {
		return err
	}
	return m.settings.SetBackupFetched(true)
}
```

We request a full month of messages for the `contact-discovery-<public-key>` content topic.
But we could be smarter here:
1. Find the latest backup message and remember it's timestamp `T`
2. Fetch more messages page by page
- ignore any messages with `timestamp != T`
- stop fetching as soon as all backup messages with this timestamp are fetched. If some message was lost, we can detect it by facing a message with the previous timestamp.

I implemented this approach in `FetchCommuntiy`, as we don't really need full month there, but only the last community description. The code now allows to proceed envelopes by page and stop the request when needed.

This will fix the issue you're referencing to. And also improve the fetching logic in general, we will not need to fetch and process redundant messages.

WDYT?

Reply:
> Are we able to fetch and iterate pages from the node most recent to least recent? 

Yeap!

> That would mean if we find backed-up data in the latest page, we don't need to iterate further in the past.

Exactly! 🙂 

> You've said Here's how we currently fetch the backup from store nodes. Do all store nodes contain the entire 30-day history?

Yes, at least that's what we assume.

Reply:
Opened an issue about this: https://github.com/status-im/status-go/issues/5022

As for this PR, I'd rather not bring back the 30 seconds delay at least as it was. 
I'll take a look, maybe (until the  https://github.com/status-im/status-go/issues/5022 is done) we could block the `Next` button for at least 30 seconds in the `ProfileFetchingState` right on the QML side. It would be much simpler to add/remove and more obvious to the user. 

Otherwise I'm on the side of not adding the delay in any form and just fix the `FetchBackup` asap in the same milestone.

Reply:
> we could block the Next button for at least 30 seconds in the ProfileFetchingState right on the QML side

With blocking on that screen user will still be able to see different total numbers for each section, depending on the order of receiving backed-up messages and that's exactly what we wanted to avoid (minimize) with that 30 seconds delay.

Reply:
Sounds good, though the timer from from 1:30 to Continue is a bit weird. Should we Show the timer starting at 30 seconds now?

Reply:
@jrainville 2minutes are kinda unrelated. 
We consider 2 minutes maximum waiting time (the actual timeout). And 30 seconds as minimum (to skip older backups).

I agree that it doesn't make much sense, but I'd rather not touch this anymore, as we will need to change it with the backup fetching improvements anyway. We won't have desktop side timeouts at all then.

Reply:
Ok sure

Reply:
The initial purpose of this is still missed.

What I see in the last video you provided is:
1. we're showing a loading indicator for all sections we're waiting a data for, and once the data comes we display what is fetched (how many contacts, keypairs...). In this approach if the received data do not refer to the latest backed-up message and another message comes later that says there are 4 keypairs to be fetched, then we need to change the expected total number for keypairs from 1 to 4, then later a new one which is for real the most recent message may have 2 keypairs, that should be for real the final total number of keypairs and we will change number 4 to 2. All that may look confusing to the user looking at those values while the fetching is ongoing, that's why initially we were displaying a loading screen for that 30 seconds delay.
2. another issue is, if we see that the data are fetched, and in you're video that's done with 100% success (all data received), why do we keep the user waiting? That's not good. So as soon as we fetch profile-related data, we should display "Continue" button (cause the profile data are mandatory only). But then if we let him proceed to the app, we're not sure if the details we've displayed in the fetching screen are the correct ones (referring to the latest backed-up message) and the user may see in the fetching screen like 10 contacts, but when lands in the app see 5 or 30 contacts, cause some new backed-up messages might be received in meantime. So from this point we're getting back to the point 1. where we need to perform that 30 second delay while the UI is displaying a loading screen, not a fetching screen.

Reply:
> The initial purpose of this is still missed.

@saledjenic, are you strictly against the new approach, considering that it's temporary anyway and will be fixed with https://github.com/status-im/status-go/issues/5022 in next milestone?

We can jump into discussion, but I really think the problem is not that big and is not worth so much time spent.

I think we should take what we have and just go further. Seeing jumping numbers during a rarely used profile fetching process won't hurt the user in anyway.

Reply:
> @saledjenic, are you strictly against the new approach, considering that it's temporary anyway and will be fixed with https://github.com/status-im/status-go/issues/5022 in next milestone?

@igor-sirotin no, far from that, I am not against it at all if we're going to fix all that via another issue.

At the end of the day, it's more a question for designers and also related Figma should be updated, now we have 2 Figma files (https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?node-id=13653%3A449718&mode=dev and https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?node-id=1024%3A298187&mode=dev) for the Onboarding part of the app and they diverge, then we have the desktop app implementation that divagates as well and also a mobile app that has different onboarding path. 

All I want is to have a single Figma file that covers the Onboarding flows and a single app implementation that follows Figma. :)

You have my approval.

Reply:
Thank you, @saledjenic!
I appreciate your input and will make sure that this screen works well after fetching backup is improved 👌